src/load/azext_load/data_plane/load_test_run/custom.py

```diff
@@ -132,6 +133,86 @@ def stop_test_run(cmd, load_test_resource, test_run_id, resource_group_name=None
     return response
 
 
+def copy_test_run_artifacts_url(cmd, load_test_resource, test_run_id, resource_group_name=None):
+    client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.info("Fetching test run copy artifacts SAS URL for test run %s", test_run_id)
+    test_run_data = client.get_test_run(test_run_id=test_run_id)
+    artifacts_container = test_run_data.get(
+        "testArtifacts", {}).get(
+        "outputArtifacts", {}).get(
+        "artifactsContainerInfo")
+    if artifacts_container is None:
+        logger.warning("No test artifacts container found for test run %s", test_run_id)
+    else:
+        logger.info("Fetched test run copy artifacts SAS URL %s", artifacts_container)
+        return artifacts_container.get("url")
+
+
+def _download_results_file(test_run_output_artifacts, test_run_id, path):
+    logger.info("Downloading results file for test run %s", test_run_id)
+    if test_run_output_artifacts is not None:
+        result_file_info = test_run_output_artifacts.get("resultFileInfo")
+        if result_file_info is not None and result_file_info.get("url") is not None:
+            file_path = get_file_info_and_download(result_file_info, path)
+            logger.warning("Results file downloaded to %s", file_path)
+        else:
+            logger.info("No results file found for test run %s", test_run_id)
+    else:
+        logger.warning(
+            "No results file found for test run %s",
+            test_run_id,
+        )
+
+
+def _download_reports_file(test_run_output_artifacts, test_run_id, path):
+    logger.info("Downloading report file for test run %s", test_run_id)
```

Comment:
nit: these methods are doing similar action can we have one method `_download_artifact_file` and remove the duplicate code

Reply:
I have refactored as much as possible to reduce duplicate code, the objects that need to be read from parent artifacts object is different in each, the logs which need to be given out are different.